cli/src/semgrep/core_runner.py

```diff
@@ -908,8 +908,16 @@ def _run_rules_direct_to_semgrep_core(
                     parsing_data.add_error(err)
             errors.extend(parsed_errors)
 
-        os.remove(rule_file_name)
-        os.remove(target_file_name)
+        try:
+            # We use NamedTemporaryFile for the rules and targets files to
+            # create files with random names, but because of how we open
+            # the files we need to remove them ourselves. However, we are
+            # not guaranteed that a different implementation would not
+            # remove the file. Therefore, we wrap this in a try-except
+            os.remove(rule_file_name)
+            os.remove(target_file_name)
```

Comment:
note that this assumes both files would be deleted together

Reply:
Yeah, might be nice to put these into separate try/except blocks so that the `logger.warning` line could be more specific about which file(s) was not found.